### PR TITLE
Support all detector modes

### PR DESCRIPTION
**Pull request by CoreyEWood**
Add support to the edge endpoint for all detector modes. Adds some basic tests to ensure responses are being parsed correctly.

### CODE REVIEW

File: app/core/edge_inference.py [hunk 1]
```diff
@@ -62,14 +62,48 @@ def parse_inference_response(response: dict) -> dict:
...
+    # Attempt to extract rois / text
+    if secondary_predictions is not None:
+        roi_predictions: list[list[dict]] | None = secondary_predictions.get("roi_predictions", None)
+        text_predictions: list[str] | None = secondary_predictions.get("text_predictions", None)
+        if roi_predictions is not None:
+            rois = roi_predictions[0]
+            for i, roi in enumerate(rois):
+                geometry = rois[i]["geometry"]
+                # TODO add validation to calculate x and y automatically
+                x = 0.5 * (geometry["left"] + geometry["right"])
```
Comment (robotrapta) [inline app/core/edge_inference.py:96]: Why are we calculating the centers here?

File: app/core/edge_inference.py [hunk 2]
```diff
@@ -62,14 +62,48 @@ def parse_inference_response(response: dict) -> dict:
...
-    }
-    output_dict["label"] = "NO" if output_dict["label"] else "YES"  # map false / 0 to "YES" and true / 1 to "NO"
+
+    multi_predictions: dict = response.get("multi_predictions", None)
+    predictions: dict = response.get("predictions", None)
+    secondary_predictions: dict = response.get("secondary_predictions", None)
+
+    if multi_predictions is not None and predictions is not None:
+        raise ValueError("Got result with both multi_predictions and predictions.")
+    if multi_predictions is not None:
```
Comment (robotrapta) [inline app/core/edge_inference.py:72]: What's a "multi_prediction"?

**Review by robotrapta — APPROVED** [review]
Looks about right to me.

File: app/core/edge_inference.py (hunk 1 above)
Comment (CoreyEWood) [reply app/core/edge_inference.py:96]: The `ImageQuery` model from the SDK requires `x` and `y` to be specified. Tyler thinks there's a way to get pydantic to do this automatically, but for now this seems easiest.

File: app/core/edge_inference.py (hunk 2 above)
Comment (CoreyEWood) [reply app/core/edge_inference.py:72]: That's the field where multiclass/counting results are returned through.

File: app/core/edge_inference.py (hunk 1 above)
Comment (tyler-romero) [reply app/core/edge_inference.py:96]: Yeah apparently Avi asked for the sdk to return the centers of the bboxes as well as left, top, right, bottom coordinates. Tim likes this too.

**Review by tyler-romero — APPROVED** [review]

File: app/api/routes/image_queries.py [hunk 3]
```diff
@@ -186,13 +186,27 @@ async def post_image_query(
...
+                result_type = ResultTypeEnum.counting
+            elif mode == ModeEnum.MULTI_CLASS:
+                result_type = ResultTypeEnum.multi_classification
+            else:
+                raise ValueError(f"Got unrecognized detector mode: {mode}")
+
             image_query = create_iqe(
                 detector_id=detector_id,
+                result_type=result_type,
                 label=results["label"],
```
Comment (brandon-groundlight) [inline app/api/routes/image_queries.py:203]: This is going to change soon, you'll need to populate this field according to the mode type

File: app/core/utils.py [hunk 4]
```diff
@@ -25,15 +28,15 @@ def create_iqe(
         created_at=datetime.utcnow(),
         query=query,
         detector_id=detector_id,
-        result_type=ResultTypeEnum.binary_classification,
+        result_type=result_type,
         result=BinaryClassificationResult(
```
Comment (brandon-groundlight) [inline app/core/utils.py:32]: We'll need to use a different type here for counting modes

**Review by brandon-groundlight — COMMENTED** [review]
I'm late to the party, this is great support for secondary predictions, but we'll eventually need to return to this to fully support the detector mode architecture

**Pull request merged**